From f943e60c8443c3f17913089898221bf99d8b3de2 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Mon, 17 Jun 2024 14:20:27 +0200
Subject: [PATCH] media: stm32: dcmipp: capture ISP stats on either P1VSYNC or
 P2VSYNC

ISP block is being used with either (or both) main & aux pipes.
Depending on which pipe is enabled, either P1VSYNC or P2VSYNC
signals will be updated.  If both pipes are running together, both
P1VSYNC & P2VSYNC will move synchronously.
Ensure to capture stats on both P1VSYNC & P2VSYNC.

Change-Id: Id9fd930d73d0cdd6f1af26e8189c30f86d7991e7
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/387987
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-statcap.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-statcap.c b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-statcap.c
index 610a85b022bc..6016ea30fb00 100644
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-statcap.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-statcap.c
@@ -22,6 +22,7 @@
 #include "dcmipp-common.h"
 
 #define DCMIPP_CMSR2_P1VSYNCF	BIT(18)
+#define DCMIPP_CMSR2_P2VSYNCF	BIT(26)
 
 #define DCMIPP_P1BPRSR			0x828
 #define DCMIPP_P1BPRSR_BADCNT_MASK	GENMASK(11, 0)
@@ -635,7 +636,8 @@ static irqreturn_t dcmipp_statcap_irq_thread(int irq, void *arg)
 		return IRQ_HANDLED;
 
 	/* We are only interested in VSYNC interrupts */
-	if (!(ved->cmsr2 & DCMIPP_CMSR2_P1VSYNCF))
+	if (!(ved->cmsr2 & DCMIPP_CMSR2_P1VSYNCF) &&
+	    !(ved->cmsr2 & DCMIPP_CMSR2_P2VSYNCF))
 		return IRQ_HANDLED;
 
 	spin_lock_irq(&vcap->irqlock);
-- 
2.39.5

